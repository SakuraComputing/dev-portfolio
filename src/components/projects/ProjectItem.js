import React from 'react';

const ProjectItem = (props) => {
    return (
        <div className="item row">
            <a className="col-md-4 col-sm-4 col-xs-12"
               href={props.project.exampleURL}
               target="_blank">
                <img className="img-responsive project-image" src={props.project.imagePath}
                     alt="project name"/>
            </a>
            <div className="desc col-md-8 col-sm-8 col-xs-12">
                <h3 className="title"><a
                    href={props.project.exampleURL}
                    target="_blank">{props.project.title}</a></h3>
                <p>{props.project.description1}</p>
                <p><a className="more-link"
                      href={props.project.exampleURL}
                      target="_blank"><i className="fa fa-external-link"></i> Find out more</a></p>
            </div>
        </div>
    );
};

export default ProjectItem;
