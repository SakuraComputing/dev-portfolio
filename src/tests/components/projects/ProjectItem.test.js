import React from 'react';
import { shallow } from 'enzyme';
import ProjectItem from '../../../components/projects/ProjectItem';

describe('Project Item', () => {

    let wrapper, project;

    beforeEach(() => {
        project = jest.fn();
        wrapper = shallow(<ProjectItem project={project}/>);
    });

    it('should render correctly', () => {
        expect(wrapper ).toMatchSnapshot();
    });
});