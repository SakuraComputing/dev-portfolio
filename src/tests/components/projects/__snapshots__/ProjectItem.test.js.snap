// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Item should render correctly 1`] = `
<div
  className="item row"
>
  <a
    className="col-md-4 col-sm-4 col-xs-12"
    target="_blank"
  >
    <img
      alt="project name"
      className="img-responsive project-image"
    />
  </a>
  <div
    className="desc col-md-8 col-sm-8 col-xs-12"
  >
    <h3
      className="title"
    >
      <a
        target="_blank"
      />
    </h3>
    <p />
    <p>
      <a
        className="more-link"
        target="_blank"
      >
        <i
          className="fa fa-external-link"
        />
         Find out more
      </a>
    </p>
  </div>
</div>
`;
